docs/module-contribution-guide.md

```diff
@@ -0,0 +1,332 @@
+
+### The `build-config` directory [Required]
+
+This directory contains the build configurations for the module. The following structure should be maintained in this directory.
+
+```shell
+build-config
+├── checkstyle
+└── resources
+```
+
+#### The `checkstyle` directory [Optional]
+
+This directory contains the checkstyle configurations for the module. This is required only if a module has Java (native) code. It includes a `build.gradle` file.
+
+### The `ballerina-tests` directory [Gradle Submodule][Optional]
+
+This directory contains the Ballerina tests that cannot be included in the `ballerina` directory (e.g. the tests that are required to be run with the compiler plugin). It also includes a `build.gradle` file, which is used to build and run the `ballerina-tests` submodule.
+
+### The `compiler-plugin` directory [Gradle Submodule][Optional]
+
+This directory contains the compiler plugin source code. It also includes a `build.gradle` file, which is used to build the `compiler-plugin` submodule.
+
+### The `compiler-plugin-tests` directory [Gradle Submodule][Optional]
+
+This directory contains the compiler plugin tests written using [TestNG](https://testng.org/doc/). It also includes a `build.gradle` file, which is used to build and run the compiler-plugin-tests submodule. This is required only if the module has a compiler plugin.
+
+### The `docs` directory [Optional]
+
+This directory contains the module specifications. It may include a `proposals` directory to add the implemented proposals.
+
+### The `examples` directory [Gradle Submodule][Optional]
+
+This directory contains the examples for the module. It includes a `build.gradle` file, which is used to build the `examples` submodule.
+
+### The `native` directory [Gradle Submodule][Optional]
+
+This directory contains the Java native code of the module. This is required only if the module has native code. 
+It also includes a `build.gradle` file, which is used to build the `native` submodule.
+
+### Other build files
+
+Apart from the above-mentioned directories, there are other files that are required for the build. These files are
+required to be added to the root directory of the module.
+
+Following are the files that are required for the build.
+```
+.
+├── LICENSE
+├── README.md
+├── build.gradle
+├── gradle.properties
+├── gradlew
+├── gradlew.bat
+├── settings.gradle
+├── spotbugs-exclude.xml
+```
+
+### The `LICENSE` file [Required]
+
+This file contains the license of the module. All the Ballerina Standard Library modules use the `Apache-2` license.
+
+### The `README.md` file [Required]
+
+This file contains the description and the build steps of the module. Usually the content of this file is similar to the `Module.md` and `Package.md` files of the Ballerina module.
+
+It also includes the status badges for the module including the `Build`, `CodeCoverage`, `Trivy`, `GraalVM`, `last commit`, and the `Open Issues` badges.
```

Comment:
We don't have all these badges in the extended libraries, right?

Reply:
Yes we have a separate set of labels. Updated in https://github.com/ballerina-platform/ballerina-extended-library/pull/505/commits/a735a8d72fe43bbbd39ad075d28ba5c396bb2b67